lucene/core/src/java/org/apache/lucene/util/automaton/FrozenIntSet.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.lucene.util.automaton;
+
+public final class FrozenIntSet extends IntSet {
+  final int state;
+
+  public FrozenIntSet(int[] values, int hashCode, int state) {
+    this.values = values;
+    this.hashCode = hashCode;
+    this.state = state;
+  }
+
+  public FrozenIntSet(int num, int state) {
+    this.values = new int[] { num };
+    this.state = state;
+    this.hashCode = 683 + num;
```

Comment:
This is a really bad hash, to be honest.... What is this constructor even used for? It's really sneaky as it allows changes to the content of a "frozen" set of values?

Reply:
It's a shortcut for the other constructor, uses the same calculation for hash that is in SortedIntSet.calculateHash, but specialized down to a single value. I'll see if we can easily drop this code since I agree that it adds complexity for maintaners.

Reply:
oh, ok. drop it; the less code to understand the better. I don't think it'll be a particular gain here.